src/frontend/src/expr/function_call.rs

```diff
@@ -107,6 +107,7 @@ impl FunctionCall {
                 Ok(DataType::Boolean)
             }
             ExprType::Coalesce => Ok(inputs[0].return_type()),
+            ExprType::ConcatWs => Ok(DataType::Varchar), // TODO: Validate input types, see below on how to recursively do so.
```

Comment:
I should have left some comments here... The `infer_type` can only handle functions with a limited number of parameters, while the ones listed here (case-when, in-list, coalesce, concat) are all variadic.

To insert cast expr over all input expr to string/varchar, call `cast_explicit` on them.
* It can be a little bit confusing as the cast is actually implicit. But casting to varchar implicitly is disallowed in general cases and `concat` is just special here.
* As you may noticed in #112, casting from another type to string has not been implemented in backend except for boolean.